plugins/inputs/internet_speed/internet_speed.go

```diff
@@ -80,31 +85,38 @@ func (is *InternetSpeed) Gather(acc telegraf.Accumulator) error {
 		"server_id": is.server.ID,
 		"host":      is.server.Host,
 	}
-	// recycle the detailed data of each test to prevent data backlog
+	// Recycle the history of each test to prevent data backlog.
 	is.server.Context.Reset()
 	acc.AddFields(measurement, fields, tags)
 	return nil
 }
 
 func (is *InternetSpeed) findClosestServer() error {
-	user, err := speedtest.FetchUserInfo()
+	var client = speedtest.New(speedtest.WithUserConfig(&speedtest.UserConfig{
+		UserAgent:  fmt.Sprintf("Telegraf SpeedTestGO/%s", speedtest.Version()),
+		ICMP:       os.Geteuid() == 0 || os.Geteuid() == -1,
+		SavingMode: is.MemorySavingMode,
+	}))
+	client.SetNThread(10)
```

Comment:
Yes, it works well, but I also think this is not a good way. Because 10 connections on the large bandwidth host may not get a value which close to the ideal bandwidth.

Reply:
Your concern is with high-bandwidth situations where they might need more threads? Why not make this also a user-configurable option then?